### PR TITLE
feat: implement gitlab release webhook

### PR DESCRIPTION
**Pull request by xNok**
https://docs.gitlab.com/ee/user/project/integrations/webhook_events.html#release-events

### CODE REVIEW

File: gitlab/gitlab_test.go [hunk 1]
```diff
@@ -94,23 +94,27 @@ func TestBadRequests(t *testing.T) {
 	for _, tt := range tests {
 		tc := tt
 		client := &http.Client{}
-		t.Run(tt.name, func(t *testing.T) {
```
Comment (robinlieb) [inline gitlab/gitlab_test.go:97]: Could you please revert this change and the indentation of the following lines?

File: gitlab/gitlab_test.go [hunk 2]
```diff
@@ -249,37 +253,50 @@ func TestWebhooks(t *testing.T) {
...
+			headers: http.Header{
+				"X-Gitlab-Event": []string{"Release Hook"},
+			},
+		},
 	}
 
 	for _, tt := range tests {
 		tc := tt
 		client := &http.Client{}
-		t.Run(tt.name, func(t *testing.T) {
```
Comment (robinlieb) [inline gitlab/gitlab_test.go:257]: Could you please revert this change and the indentation of the following lines?

File: gitlab/gitlab_test.go [hunk 3]
```diff
@@ -306,32 +323,36 @@ func TestJobHooks(t *testing.T) {
...
```
Comment (robinlieb) [inline gitlab/gitlab_test.go:309]: Could you please revert this change and the indentation of the following lines?

File: gitlab/gitlab_test.go [hunk 4]
```diff
@@ -485,31 +506,35 @@ func TestSystemHooks(t *testing.T) {
...
```
Comment (robinlieb) [inline gitlab/gitlab_test.go:488]: Could you please revert this change and the indentation of the following lines?

**Review by robinlieb — CHANGES_REQUESTED** [review]
Generally looks good, could you please revert the mentioned changes?

**Comment by xNok** [comment]
Sorry kinda forgot about this one, I reverted the indentation changes in the test

**Review by robinlieb — APPROVED** [review]

**Comment by robinlieb** [comment]
@xNok Thanks!

**Pull request merged**